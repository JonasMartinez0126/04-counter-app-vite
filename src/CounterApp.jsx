import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ name, value }) => {
  // cero es para inicializar el useState
  // counter es el valor
  // setcounter donde se realiza el cambia el valor de counter
  // cuando cambia el estado el componente vuelve a ejecutarse
  const [counter, setcounter] = useState(value);

  const handleAdd = () => {
    // el estado del counter cambio
    setcounter( counter + 1 );
  }

  const handleSubs = () => setcounter(counter - 1);
  
  const handleReset = () => setcounter(value);
  

  return (
    <>
        <h1> { name } </h1>
        <h2>{ counter }</h2>

        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSubs }>-1</button>
        <button onClick={ handleReset }>Reset</button>
    </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
}

CounterApp.defaultProps = {
    value: 123,
    name: 'Jonas Martinez'
}
  