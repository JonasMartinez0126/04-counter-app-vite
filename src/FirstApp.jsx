import React from 'react';
import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, nombre  }) => {
  return (
    <>
        <h1 data-testid="test-title" >{ title }</h1>
        <p>{ subtitle }</p>
        <p>{ subtitle }</p>
        <p>{nombre}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  title: 'No hay titulo',
  subtitle: 123,
  nombre: 'Jonas Martinez'
}