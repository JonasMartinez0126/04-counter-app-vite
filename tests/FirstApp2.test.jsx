/**  * @jest-environment jsdom  */

import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirtsApp />', () => { 

    const title = 'Hola, Soy Goku';
    const subtitle = 'Hola soy un subtitle';

    test('debe de hacer match con el snapshot', () => { 
        const { container } = render(<FirstApp title={ title } />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, Soy Goku"', () => {  
        render(<FirstApp title={title} /> );
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar el titulo en h1', () => {  
        render(<FirstApp title={title} /> );
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe de mostrar el sutitulo enviado por props', () => {  
        render(<FirstApp title={title}  subtitle={subtitle}/> );
        expect(screen.getAllByText(subtitle).length).toBe(2);
    })
})