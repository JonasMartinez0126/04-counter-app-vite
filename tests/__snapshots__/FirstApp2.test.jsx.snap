// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirtsApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
  <p>
    Jonas Martinez
  </p>
</div>
`;
